src/components/DatePicker/DatePicker.tsx

```diff
@@ -10,14 +10,42 @@ import { DatePickerTrigger } from './DatePickerTrigger';
 import { IconCaretLeft, IconCaretLeftDouble, IconCaretRight, IconCaretRightDouble } from '@foundation/Icon';
 import { Validation } from '@utilities/validation';
 
+type SingleDatePickerProps = {
+    variant?: 'single';
+    onChange: (date: Date | null) => void;
+    selectsRange?: false;
+    startDate?: null;
+    endDate?: null;
+};
+
+type RangeDatePickerProps = {
+    variant: 'range';
+    onChange: (date: [Date | null, Date | null] | null) => void;
+    selectsRange: true;
+    startDate: Date | null;
+    endDate: Date | null;
+};
+
 export type DatePickerProps = {
     placeHolder?: string;
     isClearable?: boolean;
     shouldCloseOnSelect?: boolean;
-    onChange?: (date: Date | null) => void;
     dateFormat?: string;
-    value?: Date;
+    value?: Date | null;
+    minDate?: Date;
```

Comment:
What the decision behind using minDate and maxDate properties instead of having the restriction property as `Past | Future | Past&Future | None` ?
What's the default? 
As a side note: I also notice a bug but that was already there apparently on beta, past date are greyed out but you can still select them.

Reply:
This option makes it more flexible. For example, in Asset Audit Trail we want to limit the selectable dates so that user cannot select anything before the date when the tracking started. This cannot be accomplished without setting the dates explicitly. If they're not specified, then any date is selectable